test/tests/pbs_smoketest.py

```diff
@@ -517,6 +517,17 @@ def test_submit_job_with_script(self):
         j.create_script('sleep 120\n', hostname=self.server.client)
         jid = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.logger.info("Testing script with extension")
+        j = Job(TEST_USER)
+        fn = self.du.create_temp_file(suffix=".scr", body="/bin/sleep 10",
+                                      asuser=str(TEST_USER))
+        try:
+            jid = self.server.submit(j, script=fn)
+        except PbsSubmitError as e:
+            self.assertFalse("illegal -N value" in e.msg[0],
+                             'qsub: Not accepted "." in job name')
+        else:
+            self.logger.info('Job submitted successfully: ' + jid)
```

Comment:
extend the test to make sure job is queued.

Reply:
Thanks @anamikau. Sure will extend this.

Reply:
Thanks for fixing this quickly. What I meant is to check that job is queued but a job might not be in Q state. It could be in Q|R state. you can do one of following.

self.server.expect(JOB, {'job_state': 'Q'|'R'}, id=jid)
or
self.server.expect(JOB, 'queue', id=jid)

Reply:
my bad @SudeshnaMoulik I gave you the wrong syntax. please use "{'job_state': (MATCH_RE, '[RQ]')"